control-plane/api-gateway/controllers/gatewayclass_controller.go

```diff
@@ -0,0 +1,101 @@
+package controllers
+
+import (
+	"context"
+	"time"
+
+	"github.com/go-logr/logr"
+	ctrl "sigs.k8s.io/controller-runtime"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	gwv1beta1 "sigs.k8s.io/gateway-api/apis/v1beta1"
+)
+
+const (
+	GatewayClassFinalizer      = "gateway-exists-finalizer.networking.x-k8s.io"
+	GatewayClassControllerName = "hashicorp.com/consul-api-gateway-controller"
+)
+
+// GatewayClassReconciler reconciles a GatewayClass object.
+// The GatewayClass is responsible for defining the behavior of API gateways
+// which reference the given class.
+type GatewayClassReconciler struct {
+	ControllerName string
+	Log            logr.Logger
+	client.Client
+}
+
+// +kubebuilder:rbac:groups=gateway.networking.k8s.io,resources=gatewayclasses,verbs=get;list;watch;create;update;patch;delete
```

Comment:
Just wondering -- so this is usually used for generating RBAC permissions for a controller deployment using kubebuilder, but is our helm chart actually using these or is it manually updated? If it's manual, I think we can drop these annotations.

Reply:
This is an absolute cargo cult scenario. I didn't really know what these did, but I included them in case I needed them.